### PR TITLE
update CONTRIBUTING to be on par with the NeIC handbook

### LINKED ISSUE #692: Update the contribution guidelines
The file should be on par with what's in the handbook.

### PR DESCRIPTION
**Pull request by aaperis**
**Related issue(s) and PR(s)**  
This PR closes #692 .

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,59 +1,58 @@
...
 
-  - a *clear* description,
-  - as much relevant information as possible, and
-  - a *code sample* or an (executable) *test case* demonstrating the expected behaviour that is not occurring.
+- If you're unable to find an (open) issue addressing the problem, open a new one by using the following [template to report a bug]. Be sure to include:
 
-* If possible, use the following [template to report a bug].
+      - a *clear* description,
+      - as much relevant information as possible, and
+      - a *code sample* or an (executable) *test case* demonstrating the expected behaviour that is not occurring.
```
Comment (MalinAhlberg) [inline CONTRIBUTING.md:13]:
```suggestion
    - a *clear* description,
    - as much relevant information as possible, and
    - a *code sample* or an (executable) *test case* demonstrating the expected behaviour that is not occurring.
```
To fix formatted version.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,59 +1,58 @@
...
+- Create an issue on Github or you can alternatively pick one already created.
 
-Assign yourself to that issue.
+- Assign yourself to that issue.
 
-Discussions on how to proceed about that issue take place in the comment section on that issue.
+- Discussions on how to proceed about that issue take place in the comment section on that issue.
 
-Some of the work might have been done already by somebody, hence we avoid unnecessary work duplication and a waste of time and effort. Other reason for discussing the issue beforehand is to communicate with the team the changes as some of the features might impact different components, and we can plan accordingly.
+Some of the work might have been done already by a co-worker. In this case, unnecessary work duplication and waste of time and effort are to be avoided. As some of the features might impact different components, it is a good idea to discuss the issue beforehand with the team and communicate the intended changes so that planning can be done accordingly.
```
Comment (MalinAhlberg) [inline CONTRIBUTING.md:25]:
```suggestion
To avoid unnecessary work duplication and waste of time and effort, it's generally a good idea to discuss the issue beforehand with team. Some of the work might have been done already by a co-worker. Also, as some of the features might impact different components, please communicate the intended changes, so that planning can be done accordingly.
```

Comment (MalinAhlberg) [inline CONTRIBUTING.md:25]:
```suggestion
To avoid unnecessary work duplication and waste of time and effort, it's generally a good idea to discuss the issue beforehand with team. Some of the work might have been done already by a co-worker. Also, as some of the features might impact different components, please communicate the intended changes, so that planning can be done accordingly.
```
I found this section a bit hard to follow, so re-ordered it a bit. Feel free to use och discard :)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -69,36 +68,39 @@ Some tips about writing helpful commit messages:
...
 
 
 ### How we do code reviews
 
 A code review is initiated when someone has made a Pull Request in the appropriate repo on github.
 
-Work should not continue on the branch _unless_ it is a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/). Once the PR is marked ready the review can start.
+Work should not continue on the branch *unless* it is a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/). Once the PR is marked ready the review can start.
+
+The initiator of the PR should assign the [@sensitive-data-development-collaboration](https://github.com/orgs/neicnordic/teams/sensitive-data-development-collaboration) team for reviewer duty on the branch.
```
Comment (MalinAhlberg) [inline CONTRIBUTING.md:80]:
```suggestion
The initiator of the PR should assign the [@sensitive-data-development-collaboration](https://github.com/orgs/neicnordic/teams/sensitive-data-development-collaboration) team as reviews of the branch.
```

**Review by MalinAhlberg — APPROVED** [review]
Very nice :clap: !
I added some minor suggestions.
It would be nice, in the future, to go through the contribution guidelines we have in confluence, and make sure that they match with these (maybe a new issue :smiley: ?)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,59 +1,58 @@
...
+# Contributing Guidelines
 
+We thank you in advance 👍 🎉 for taking the time to contribute, whether with *code* or with *ideas*, to the NeIC SDA project.
 
 ## Did you find a bug?
 
-* Ensure that the bug was not already reported by [searching under Issues].
-
-* If you're unable to find an (open) issue addressing the problem, [open a new one]. Be sure to prefix the issue title with **[BUG]** and to include:
+- Ensure that the bug was not already reported by [searching under Issues].
```
Comment (nanjiangshu) [inline CONTRIBUTING.md:7]:
```suggestion
- Ensure that the bug has not already been reported by [searching under Issues].
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,59 +1,58 @@
...
 
-  - a *clear* description,
-  - as much relevant information as possible, and
-  - a *code sample* or an (executable) *test case* demonstrating the expected behaviour that is not occurring.
+- If you're unable to find an (open) issue addressing the problem, open a new one by using the following [template to report a bug]. Be sure to include:
```
Comment (nanjiangshu) [inline CONTRIBUTING.md:9]:
```suggestion
- If you're unable to find an issue addressing the problem, open a new one by using the following [template to report a bug]. Be sure to include:
```

Comment (nanjiangshu) [reply CONTRIBUTING.md:9]: A bit unclear to use parenthesis (open) here. Better to change to either `an issue` (as I suggested) or `an open issue`.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,59 +1,58 @@
...
 
+Before making the pull request it is a good idea to rebase your branch to `main` to ensure that eventual conflicts with the `main` branch are solved before the PR is reviewed and that there can be a clean merge.
+> NOTE:
+> In older github repositories the default branch might be called `master` instead of `main`.
 
 
-### General stuff about git and commit messages
+### About git and commit messages
 
 In general it is better to commit often. Small commits are easier to roll back and also makes the code easier to review.
```
Comment (nanjiangshu) [inline CONTRIBUTING.md:48]:
```suggestion
In general it is better to commit often. Small commits are easier to roll back and also make the code easier to review.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,59 +1,58 @@
...
 
-Write helpful commit messages that describes the changes and possibly why they were necessary.
+Write helpful commit messages that describe the changes and possibly why they were necessary.
 
-Each commit should contain changes that are functionally connected and/or related. If you for example want to write _and_ in the first line of the commit message this is an indicator that it should have been two commits.
-
-Learn how to select chunks of changed files to do multiple separate commits of unrelated things. This can be done with either `git add -p` or `git commit -p`.
+Each commit should contain changes that are functionally connected and/or related. If, for example, the first line of the commit message contains the word *and*, this is an indicator that this commit should have been split in two.
```
Comment (nanjiangshu) [inline CONTRIBUTING.md:52]:
```suggestion
Each commit should contain changes that are functionally connected and/or related. If, for example, the first line of the commit message contains the word *and*, this is an indicator that this commit should have been split into two.
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (aaperis) [reply CONTRIBUTING.md]: Reads better this way :-)

**Review by nanjiangshu — COMMENTED** [review]
Great addition! I just added a few minor comments.

File: CONTRIBUTING.md (hunk 5 above)
Comment (aaperis) [reply CONTRIBUTING.md]: Better without the `()`.

**Comment by aaperis** [comment]
> Very nice 👏 ! I added some minor suggestions. It would be nice, in the future, to go through the contribution guidelines we have in confluence, and make sure that they match with these (maybe a new issue 😃 ?)

Indeed, revisiting this has been in the SM backlog for some time now :smile:

**Review by MalinAhlberg — APPROVED** [review]
